apiserver/provisioner/provisioner_test.go

```diff
@@ -1235,3 +1235,37 @@ func (s *withoutControllerSuite) TestFindTools(c *gc.C) {
 		c.Assert(tools.URL, gc.Equals, url)
 	}
 }
+
+func (s *withoutControllerSuite) TestMarkMachinesForRemoval(c *gc.C) {
+	err := s.machines[0].EnsureDead()
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.machines[2].EnsureDead()
+	c.Assert(err, jc.ErrorIsNil)
+
+	res, err := s.provisioner.MarkMachinesForRemoval(params.Entities{
+		Entities: []params.Entity{
+			{Tag: "machine-2"},         // ok
+			{Tag: "machine-100"},       // not found
+			{Tag: "machine-0"},         // ok
+			{Tag: "machine-1"},         // not dead
+			{Tag: "machine-0-lxd-5"},   // unauthorised
+			{Tag: "application-thing"}, // only machines allowed
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	results := res.Results
+	c.Assert(results, gc.HasLen, 6)
+	c.Check(results[0].Error, gc.IsNil)
+	c.Check(*results[1].Error, gc.Equals,
+		*common.ServerError(errors.NotFoundf("machine 100")))
```

Comment:
Explicitly checking the code (IsCodeNotFound?) is probably best here -- that's what any consumer will actually be responding to. (It's *also* nice to check the message too, but the code is the really important thing.)